### DISCUSSION TITLE
dependabot

### DISCUSSION THREAD

**Discussion by bobturneruk** [pykale/pykale#179] [Ideas] [answered]
dependabot might help us learn of potential problems caused by dependency updates so we can fix them early:

https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/about-dependabot-version-updates

I'll have a bit more of a look, but I'm interested in others' views.

**Comment by haipinglu** [comment]
It will be great to have it. Since the link is a GitHub page, it seems to be something (that can be) well integrated. Thanks.

**Comment by bobturneruk** [comment]
I note that dependabot is not detecting extra and dev dependencies. Not sure if this is configurable.

https://github.com/pykale/pykale/network/dependencies

**Comment by bobturneruk** [comment, marked as answer]
I think dependabot is also not doing exactly what I'd hoped around letting us know when new dependency version arise to break our code (e.g. https://github.com/pykale/pykale/issues/194). I suggest we specify acceptable version ranges in setup.py to avoid these surprises, in future https://stackoverflow.com/questions/11472810/how-to-specify-version-ranges-in-install-requires-setuptools-distribute
